src/utils/middleware/checkIP.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * An Express middleware that checks if source IP falls into a CIDR range.
+ */
+
+import * as express from "express";
+import { isLeft } from "fp-ts/lib/Either";
+import * as rangeCheck from "range_check";
+import * as requestIp from "request-ip";
+import { IPString } from "../strings";
+
+export default function checkIP(
+  range: string
```

Comment:
there's no check whether `range` is indeed a valid range, you should have a specialized `string` type (perhaps declared in `strings.ts`) to make sure the caller has checked that `range` is valid before calling this function